core/logbook/src/main/java/org/phoebus/logbook/LogClient.java

```diff
@@ -14,434 +15,458 @@
 public interface LogClient {
 
     /**
-     * Get a list of all the logbooks currently existings
-     * 
-     * @return string collection of logbooks
+     * Create a single LogEntry <code>log</code>, if the LogEntry already exists it is replaced.
+     *
+     * @param log - LogEntry to be added
+     * @return - the created log entry
      */
-    public Collection<Logbook> listLogbooks();
+    public LogEntry set(LogEntry log) throws LogbookException;
 
     /**
-     * Get a list of all the tags currently existing
-     * 
-     * @return string collection of tags
+     * Returns a LogEntry that exactly matches the logId <code>logId</code>
+     *
+     * @param logId LogEntry id
+     * @return LogEntry object
      */
-    public Collection<Tag> listTags();
+    public LogEntry getLog(Long logId);
 
     /**
-     * Get a list of all the Properties currently existing
-     * 
-     * @return
+     * Returns a collection of attachments that matches the logId <code>logId</code>
+     *
+     * @param logId LogEntry id
+     * @return attachments collection object
      */
-    public Collection<Property> listProperties();
+    public Collection<Attachment> listAttachments(Long logId);
 
     /**
-     * List the supported log levels
-     * 
-     * @return a list of supported levels
-     */
-    public Collection<String> listLevels();
-    
-    /**
-     * List all the active attributes associated with the property
-     * <tt>propertyName</tt> property must exist, name != null
-     * 
-     * @param propertyName
-     * @return
+     * Query for logs based on the criteria specified in the map
+     *
+     * @param map - search parameters
+     * @return collection of LogEntry objects
      */
-    public Collection<String> listAttributes(String propertyName);
+    public List<LogEntry> findLogs(Map<String, String> map);
 
     /**
      * Return all the logs. ***Warning can return a lot of data***
-     * 
+     *
      * @return Collection of all LogEntry entires
```

Comment:
Several comments use "entires" instead of "entries"

Reply:
sorry, other that this one the other 'logentries' and 'logentires' both appears as a spelling mistake so I landed up ignoring them.
Fixed and creating PR now